lib/internal/process/execution.js

```diff
@@ -171,7 +171,10 @@ function createOnGlobalUncaughtException() {
     // call that threw and was never cleared. So clear it now.
     clearDefaultTriggerAsyncId();
 
-    const type = fromPromise ? 'unhandledRejection' : 'uncaughtException';
+    let type = fromPromise ? 'unhandledRejection' : 'uncaughtException';
+    // Redefine the type only when it's an UnhandledErrorEvent.
+    // Refs: https://github.com/nodejs/node/issues/51202
+    if (er.name === 'UnhandledErrorEvent') type = 'unhandledErrorEvent';
```

Comment:
and then this could use instanceof instead of the much less robust name check

Reply:
True that